examples/dtls-echo/dtls-server.c

```diff
@@ -0,0 +1,449 @@
+}
+
+static int peer_verify_ecdsa_key(struct dtls_context_t *ctx,
+                 const session_t *session,
+                 const unsigned char *other_pub_x,
+                 const unsigned char *other_pub_y,
+                 size_t key_size)
+{
+    (void) ctx;
+    (void) session;
+    (void) other_pub_x;
+    (void) other_pub_y;
+    (void) key_size;
+    return 0;
+}
+#endif /* DTLS_ECC */
+
+/**
+ * @brief We prepare the DTLS for this node.
+ */
+static void init_dtls(void)
+{
+    static dtls_handler_t cb = {
+        .write = send_to_peer,
+        .read  = read_from_peer,
+        .event = NULL,
+#ifdef DTLS_PSK
+        .get_psk_info = peer_get_psk_info,
+#endif  /* DTLS_PSK */
+#ifdef DTLS_ECC
+        .get_ecdsa_key = peer_get_ecdsa_key,
+        .verify_ecdsa_key = peer_verify_ecdsa_key
+#endif  /* DTLS_ECC */
+    };
+
+
+#ifdef DTLS_PSK
+    puts("Server support PSK");
+#endif
+#ifdef DTLS_ECC
+    puts("Server support ECC");
+#endif
+
+    DEBUG("DBG-Server On\n");
+
+    /*
+     * The context for the server is a little different from the client.
+     * The simplicity of GNRC do not mix transparently with
+     * the DTLS Context. At this point, the server need a fresh context
+     * however dtls_context->app must be populated with an unknown
+     * IPv6 address.
+     *
+     * The non-valid Ipv6 address ( :: ) is discarded due the chaos.
+     * For now, the first value will be the loopback.
+     */
+    char *addr_str = "::1";
+
+    /*akin to syslog: EMERG, ALERT, CRITC, NOTICE, INFO, DEBUG */
+    dtls_set_log_level(DTLS_LOG_DEBUG);
+
+
+    dtls_context = dtls_new_context(addr_str);
+    if (dtls_context) {
+        dtls_set_handler(dtls_context, &cb);
+    }
+    else {
+        puts("Server was unable to generate DTLS Context!");
+        exit(-1);
+    }
+
+
+
+}
+
+/* NOTE: wrapper or trampoline ? (Syntax question) */
+
+void *dtls_server_wrapper(void *arg)
+{
+    (void) arg; /* TODO: Remove? We don't have args at all (NULL) */
+
+    msg_t _reader_queue[READER_QUEUE_SIZE];
+    msg_t msg;
+
+    /* The GNRC examples uses packet dump but we want a custom one */
+    msg_init_queue(_reader_queue, READER_QUEUE_SIZE);
+
+    init_dtls();
+
+    /*
+     * FIXME: After mutliple retransmissions, and canceled client's sessions
+     * the server become unable to sent NDP NA messages. Still, the TinyDTLS
+     * debugs seems to be fine.
+     */
+
+    while (1) {
+
+        /* wait for a message */
+        msg_receive(&msg);
+
+        DEBUG("DBG-Server: Record Rcvd!\n");
+        dtls_handle_read(dtls_context, (gnrc_pktsnip_t *)(msg.content.ptr));
+
+        /*TODO: What happens with other clients connecting at the same time? */
+
+    } /*While */
+
+    dtls_free_context(dtls_context);
+}
+
+static void start_server(void)
+{
+    uint16_t port;
+
+    port = (uint16_t)DEFAULT_PORT;
+
+    (void) _dtls_kernel_pid;
+
+    /* Only one instance of the server */
+    if (server.pid != KERNEL_PID_UNDEF) {
```

Comment:
> and this PR needs a rebase to current master

Just to avoid the conflicts from the previous week.  This means to fetch my local repository with the changes from RIOT-OS and then rebase my branch with those changes, right ?